### PR TITLE
Accessiblity

### PR DESCRIPTION
**Pull request by pavpanchekha**

### CODE REVIEW

File: book/accessibility.md [hunk 1]
```diff
@@ -324,11 +324,13 @@ class BlockLayout:
     	# ...
         px_size = float(node.style["font-size"][:-2])
         size = dpx(px_size * 0.75, self.zoom)
+    	# ...
```
Comment (pavpanchekha) [inline book/accessibility.md:327]: It's not the last line in the file

File: book/accessibility.md [hunk 2]
```diff
@@ -1482,10 +1486,10 @@ class LineLayout:
 ```
 
 For the [focus example](examples/example14-focus.html), the focus outline
-of an `<a>` element becomes red, as in Figure 6.
+of an `<a>` element becomes thicker and red, as in Figure 6.
```
Comment (pavpanchekha) [inline book/accessibility.md:1490]: The red isn't visible in the book, but "thicker" is and they'll hopefully trust us on "red"

File: book/accessibility.md [hunk 3]
```diff
@@ -1625,7 +1629,6 @@ class AccessibilityNode:
     def __init__(self, node):
         self.node = node
         self.children = []
-        self.text = ""
```
Comment (pavpanchekha) [inline book/accessibility.md:1628]: Not used until later.

File: book/accessibility.md [hunk 4]
```diff
@@ -1822,8 +1825,10 @@ and [`playsound`][playsound]. You can install them using `pip`:
 
 [playsound]: https://pypi.org/project/playsound/
 
-    python3 -m pip install gtts
-    python3 -m pip install playsound
+``` {.sh}
+python3 -m pip install gtts
+python3 -m pip install playsound
+```
```
Comment (pavpanchekha) [inline book/accessibility.md:1832]: Should not be an example style.

File: book/accessibility.md [hunk 5]
```diff
@@ -2440,9 +2445,7 @@ class Browser:
 
 By the way, the acronym `a11y` in `a11y_node`, with an "a", the number 11, and
 a "y", is a common shorthand for the word "accessibility".[^why-11] The
-`hit_test` function I'm calling is the same one we wrote
-[earlier in Section 7.3](chrome.md#click-handling) to handle clicks, except
-of course that it's searching a different tree:
+`hit_test` function recurses over the accessibility tree:
```
Comment (pavpanchekha) [inline book/accessibility.md:2449]: It is not similar! An earlier draft of the book used similar code, but now the A11y hit-test method is recursive while layout hit-test isn't.

File: book/accessibility.md [hunk 6]
```diff
@@ -141,7 +141,7 @@ Zoom
 
 Let's start with the simplest accessibility problem: text on the
 screen that is too small to read. It's a problem many of us will face
-sooner or later, and possibly the most common user disability issue.
+sooner or later, and is possibly the most common user disability issue.
```
Comment (chrishtr) [inline book/accessibility.md:144]: Not critical IMO

File: book/accessibility.md [hunk 7]
```diff
@@ -279,8 +278,10 @@ class DocumentLayout:
 ``` {.python}
 class Tab:
     def render(self):
-        # ...
+        if self.needs_layout:
```
Comment (chrishtr) [inline book/accessibility.md:281]: Not critical IMO

File: book/accessibility.md (hunk 1 above)
Comment (chrishtr) [reply book/accessibility.md:328]: Not critical IMO

File: book/accessibility.md [hunk 8]
```diff
@@ -324,11 +325,13 @@ class BlockLayout:
     	# ...
         px_size = float(node.style["font-size"][:-2])
         size = dpx(px_size * 0.75, self.zoom)
+    	# ...
 
     def input(self, node):
-	    # ...
+        # ...
```
Comment (chrishtr) [inline book/accessibility.md:331]: Not critical IMO

File: book/accessibility.md [hunk 9]
```diff
@@ -337,6 +340,7 @@ class InputLayout:
         # ...
         px_size = float(self.node.style["font-size"][:-2])
         size = dpx(px_size * 0.75, self.zoom)
+    	# ...
```
Comment (chrishtr) [inline book/accessibility.md:343]: Not critical IMO

File: book/accessibility.md [hunk 10]
```diff
@@ -1199,7 +1204,7 @@ def paint_outline(node, cmds, rect, zoom):
     cmds.append(DrawOutline(rect, "black", 1))
 ```
 
-Set this flag in a new `focus_element` method that we'll now use
+Set this `is_focused` flag in a new `focus_element` method that we'll now use
```
Comment (chrishtr) [inline book/accessibility.md:1207]: Not critical IMO

File: book/accessibility.md [hunk 11]
```diff
@@ -1415,7 +1420,7 @@ Unknown pseudoclasses simply never match anything.
 The focused element can now be styled. But ideally we'd also be able to
 customize the focus outline itself and not just the element. That can be done
 by adding support for the CSS [`outline` property][outline], which looks like
-this to show a 3px red outline:[^outline-syntax]
+this (for a 3-pixel-thick red outline):[^outline-syntax]
```
Comment (chrishtr) [inline book/accessibility.md:1423]: Not critical IMO

File: www/examples/example14-focus.html [hunk 12]
```diff
@@ -7,41 +7,8 @@
...
-.
-<br>
-.
-<br>
-.
-<br>
-.
-<br>
-.
+<br> . <br> . <br> . <br> . <br> . <br> .
```
Comment (chrishtr) [inline www/examples/example14-focus.html:10]: Not critical IMO, but OTOH it does make the proof look a lot better...

**Review by chrishtr — APPROVED** [review]

**Pull request merged**